src/axom/primal/geometry/CoordinateTransformer.hpp

```diff
@@ -0,0 +1,450 @@
+    P[0][1] = u[0] * u[1] * ccosT - u[2] * sinT;
+    P[0][2] = u[0] * u[2] * ccosT + u[1] * sinT;
+    P[1][0] = u[1] * u[0] * ccosT + u[2] * sinT;
+    P[1][1] = u[1] * u[1] * ccosT + cosT;
+    P[1][2] = u[1] * u[2] * ccosT - u[0] * sinT;
+    P[2][0] = u[2] * u[0] * ccosT - u[1] * sinT;
+    P[2][1] = u[2] * u[1] * ccosT + u[0] * sinT;
+    P[2][2] = u[2] * u[2] * ccosT + cosT;
+
+    // Multiply P*m_P, saving in pNew, then copy back to m_P.
+    Matrx pNew = {{0, 0, 0}, {0, 0, 0}, {0, 0, 0}};
+    for(int r = 0; r < 3; ++r)
+    {
+      for(int c = 0; c < 3; ++c)
+      {
+        for(int k = 0; k < 3; ++k)
+        {
+          pNew[r][c] += P[r][k] * m_P[k][c];
+        }
+      }
+    }
+    for(int r = 0; r < 3; ++r)
+    {
+      for(int c = 0; c < 3; ++c)
+      {
+        m_P[r][c] = pNew[r][c];
+      }
+    }
+
+    // Change to m_v.
+    Vectr vOld = m_v;
+    multMatrxVectr(P, vOld, m_v);
+  }
+
+  //! @brief Get a transformed 3D Point.
+  AXOM_HOST_DEVICE axom::primal::Point<T, 3> getTransformed(const axom::primal::Point<T, 3>& pt) const
+  {
+    axom::primal::Point<T, 3> rval = pt;
+    transform(rval[0], rval[1], rval[2]);
+    return rval;
+  }
+
+  //! @brief Get a transformed 3D coordinate.
+  AXOM_HOST_DEVICE axom::NumericArray<T, 3> getTransformed(const axom::NumericArray<T, 3>& pt) const
+  {
+    axom::NumericArray<T, 3> rval = pt;
+    transform(rval[0], rval[1], rval[2]);
+    return rval;
+  }
+
+  //! @brief Transform a 3D coordinate in place.
+  AXOM_HOST_DEVICE void transform(axom::NumericArray<T, 3>& pt) const
+  {
+    transform(pt[0], pt[1], pt[2]);
+  }
+
+  //! @brief Transform a 3D coordinate in place.
+  AXOM_HOST_DEVICE void transform(T& x, T& y, T& z) const
+  {
+    Vectr tmpPt = {x, y, z};
+    const auto& P(m_P);  // shorthand
+    x = P[0][0] * tmpPt[0] + P[0][1] * tmpPt[1] + P[0][2] * tmpPt[2] + m_v[0];
+    y = P[1][0] * tmpPt[0] + P[1][1] * tmpPt[1] + P[1][2] * tmpPt[2] + m_v[1];
+    z = P[2][0] * tmpPt[0] + P[2][1] * tmpPt[1] + P[2][2] * tmpPt[2] + m_v[2];
+  }
+
+  /*!
+   * @brief Invert the transformation in place.
+   *
+   * Using a special inverse formula for 4x4 matrices with last row [0,0,0,1].
+   * @verbatim
+   * Minv = [ Pinv - Pinv*v ]
+   *        [  0       1    ]
+   * @endverbatim
+   */
+  AXOM_HOST_DEVICE void invert()
+  {
+    invertMatrx(m_P);
+    auto negV = -m_v;
+    multMatrxVectr(m_P, negV, m_v);
+  }
+
+  //! brief Get the inverse transformer.
+  CoordinateTransformer getInverse() const
+  {
+    auto rval = *this;
+    rval.invert();
+    return rval;
+  }
+
+private:
+  //!@brief 3-vector or 3D coordinates, depending on context.
+  using Vectr = axom::NumericArray<T, 3>;
+
+  //!@brief 3x3 transformation matrix with row-major storage.
+  using Matrx = Vectr[3];
+
+  /*
+   * The 4x4 matrix is saved as a 3x3 matrix P and a 3x1 vector v.
+   * Last row is not stored because it's always [0,0,0,1].
+   * M = [ P v ]
+   *     [ 0 1 ]
+   */
```

Comment:
Please add something like the following:
```suggestion
   *
   * Store m_P[0][0] = std::numeric_limits<T>::quiet_NaN() to set this 
   * CoordinateTransformer as invalid.
   */
```